internal/fortran/fortran.go

```diff
@@ -0,0 +1,422 @@
+	}
+}
+
+// SideCharacter returns the FORTRAN character for the given side state.
+func SideCharacter(side blas.Side) byte {
+	switch side {
+	case blas.Left:
+		return 'L'
+	case blas.Right:
+		return 'R'
+	default:
+		panic("fortran: bad BLAS side")
+	}
+}
+
+// General is a column major general matrix.
+type General blas64.General
+
+// FromRowMajorGeneral returns a column major general matrix
+// with the same dimensions and data elements as the row major a.
+func FromRowMajorGeneral(a blas64.General) General {
+	t := General{
+		Rows:   a.Cols,
+		Cols:   a.Rows,
+		Stride: a.Rows,
+		Data:   make([]float64, a.Rows*a.Cols),
+	}
+	for i := 0; i < a.Rows; i++ {
+		for j := 0; j < a.Cols; j++ {
+			t.Data[i+j*t.Stride] = a.Data[i*a.Stride+j]
+		}
+	}
+	return t
+}
+
+// ToRowMajorGeneral returns a row major general matrix
+// with the same dimensions and data elements as a column major a.
+func ToRowMajorGeneral(a General) blas64.General {
+	t := blas64.General{
+		Rows:   a.Cols,
+		Cols:   a.Rows,
+		Stride: a.Cols,
+		Data:   make([]float64, a.Rows*a.Cols),
+	}
+	for i := 0; i < a.Cols; i++ {
+		for j := 0; j < a.Rows; j++ {
+			t.Data[i+j*t.Stride] = a.Data[i*a.Stride+j]
+		}
+	}
+	return t
+}
+
+// CopyGeneralRowMajor performs a copy of src to dst which are
+// row major matrices. The dimensions of src and dst must match
+// and dst must have adequate data storage, otherwise CopyGeneralRowMajor
+// will panic.
+func CopyGeneralRowMajor(dst, src blas64.General) {
+	if dst.Rows != src.Rows || dst.Cols != src.Cols {
+		panic("fortran: mismatched dimension")
+	}
+	if len(dst.Data) < (dst.Rows-1)*dst.Stride+dst.Cols {
+		panic("fortran: short data slice")
+	}
+	for i := 0; i < src.Rows; i++ {
+		for j := 0; j < src.Cols; j++ {
+			dst.Data[i*dst.Stride+j] = src.Data[i*src.Stride+j]
+		}
+	}
+}
+
+// CopyGeneralColMajor performs a copy of src to dst which are
+// column major matrices. The dimensions of src and dst must match
+// and dst must have adequate data storage, otherwise CopyGeneralRowMajor
+// will panic.
+func CopyGeneralColMajor(dst, src General) {
+	if dst.Rows != src.Rows || dst.Cols != src.Cols {
+		panic("fortran: mismatched dimension")
+	}
+	if len(dst.Data) < (dst.Cols-1)*dst.Stride+dst.Rows {
+		panic("fortran: short data slice")
+	}
+	for j := 0; j < src.Cols; j++ {
+		for i := 0; i < src.Rows; i++ {
+			dst.Data[i+j*dst.Stride] = src.Data[i+j*src.Stride]
+		}
+	}
+}
+
+// Symmetric is a column major symmetric matrix.
+type Symmetric blas64.Symmetric
+
+// FromRowMajorSymmetric returns a column major symmetric matrix
+// with the same dimensions and data elements as the row major a.
+func FromRowMajorSymmetric(a blas64.Symmetric) Symmetric {
+	t := Symmetric{
+		N:      a.N,
+		Stride: a.N,
+		Data:   make([]float64, a.N*a.N),
+		Uplo:   a.Uplo,
+	}
+	switch a.Uplo {
+	case blas.Lower:
+		for i := 0; i < a.N; i++ {
+			for j := 0; j <= i; j++ {
+				t.Data[i+j*t.Stride] = a.Data[i*a.Stride+j]
+			}
+		}
+	case blas.Upper:
```

Comment:
Nitpicking, but in basically all of the blas/lapack code the Upper case is first.